Lib/test/test_unittest/test_case.py

```diff
@@ -346,8 +346,12 @@ class Foo(unittest.TestCase):
             async def test1(self):
                 return 1
 
-        with self.assertWarns(DeprecationWarning) as w:
-            Foo('test1').run()
+        with warnings.catch_warnings():
```

Comment:
I think neater to use check_warnings: [`f0490ab` (#128726)](https://github.com/python/cpython/pull/128726/commits/f0490abc262321749f2c487acb1fd8e7fde2df48)

Reply:
The RuntimeWarning is less important for this test.

I think that `check_warnings` is too heavy for this. It requires to repeat the DeprecationWarning message.